CombinatorialGames/Game/Reductions.lean

```diff
@@ -0,0 +1,101 @@
+/-
+Copyright (c) 2025 Aaron Liu. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Aaron Liu
+-/
+import CombinatorialGames.Game.IGame
+import Mathlib.Algebra.Group.Pointwise.Set.Lattice
+
+/-!
+# Game reductions
+
+We prove that dominated moves can be deleted and reversible moves can be bypassed.
+-/
+
+universe u
+
+namespace IGame
+
+theorem equiv_of_dominated_left {u v r : Set IGame.{u}} [Small.{u} u] [Small.{u} v] [Small.{u} r]
+    (hu : ∀ g ∈ u, ∃ g' ∈ v, g ≤ g') : {u ∪ v | r}ᴵ ≈ {v | r}ᴵ := by
+  apply equiv_of_exists_le <;> aesop
+
+theorem equiv_of_dominated_right {l u v : Set IGame.{u}} [Small.{u} l] [Small.{u} u] [Small.{u} v]
+    (hu : ∀ g ∈ u, ∃ g' ∈ v, g' ≤ g) : {l | u ∪ v}ᴵ ≈ {l | v}ᴵ := by
+  apply equiv_of_exists_le <;> aesop
+
+theorem equiv_of_bypass_left {ι : Type u} {l r : Set IGame.{u}} [Small.{u} l] [Small.{u} r]
+    {c cr : ι → IGame.{u}} (hbb : ∀ i, cr i ≤ {Set.range c ∪ l | r}ᴵ)
+    (hcr : ∀ i, cr i ∈ (c i).rightMoves) :
+    {Set.range c ∪ l | r}ᴵ ≈ {(⋃ i, (cr i).leftMoves) ∪ l | r}ᴵ := by
```

Comment:
I don't like this statement for many of the same reasons (but worse!) But I don't know how I'd restate it to make it better. Hmm...